src/iop/temperature.c

```diff
@@ -456,18 +466,37 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
   const float *const in = (const float *const)ivoid;
   float *const out = (float *const)ovoid;
+  const float *const d_coeffs = d->coeffs;
 
   if(filters == 9u)
   { // xtrans float mosaiced
 #ifdef _OPENMP
-#pragma omp parallel for SIMD() default(none) \
-    dt_omp_firstprivate(d, in, out, roi_out, xtrans) \
-    schedule(static) \
-    collapse(2)
+#pragma omp parallel for default(none) \
+    dt_omp_firstprivate(d_coeffs, in, out, roi_out, xtrans) \
+    schedule(static)
 #endif
     for(int j = 0; j < roi_out->height; j++)
     {
-      for(int i = 0; i < roi_out->width; i++)
+      const float DT_ALIGNED_PIXEL coeffs[3][4] =
+      {
+        { d_coeffs[FCxtrans(j, 0, roi_out, xtrans)], d_coeffs[FCxtrans(j, 1, roi_out, xtrans)],
+          d_coeffs[FCxtrans(j, 2, roi_out, xtrans)], d_coeffs[FCxtrans(j, 3, roi_out, xtrans)] },
+        { d_coeffs[FCxtrans(j, 4, roi_out, xtrans)], d_coeffs[FCxtrans(j, 5, roi_out, xtrans)],
+          d_coeffs[FCxtrans(j, 6, roi_out, xtrans)], d_coeffs[FCxtrans(j, 7, roi_out, xtrans)] },
+        { d_coeffs[FCxtrans(j, 8, roi_out, xtrans)], d_coeffs[FCxtrans(j, 9, roi_out, xtrans)],
+          d_coeffs[FCxtrans(j, 10, roi_out, xtrans)], d_coeffs[FCxtrans(j, 11, roi_out, xtrans)] },
+      };
+      // process sensels four at a time
+      //(note that attempting to ensure alignment for this main loop actually slowed things down marginally)
+      int i = 0;
+      for(int coeff = 0; i + 4 < roi_out->width; i += 4, coeff = (coeff+1)%3)
+      {
+        const size_t p = (size_t)j * roi_out->width + i;
+        for_four_channels(c) // in and out are NOT aligned when width is not a multiple of 4
+          out[p+c] = in[p+c] * coeffs[coeff][c];
+      }
+      // process the leftover sensels
+      for(; i < roi_out->width; i++)
       {
         const size_t p = (size_t)j * roi_out->width + i;
         out[p] = in[p] * d->coeffs[FCxtrans(j, i, roi_out, xtrans)];
```

Comment:
```suggestion
        out[p] = in[p] * d_coeffs[FCxtrans(j, i, roi_out, xtrans)];
```

because we're in parallel section and `d` got thrown out do d_coeffs it should be :)

Reply:
Missed that one because GCC8 automatically assumes `shared`  for const variables even with `default(none)` -- in fact, won't let you declare them as such in the OpenMP pragma, hence why we have dt_omp_sharedconst.